pkg/scheduler/internal/queue/scheduling_queue_test.go

```diff
@@ -1840,94 +1840,123 @@ func TestPerPodSchedulingMetrics(t *testing.T) {
+			name: "A gated pod is created and scheduled after lifting gate",
+			perPodSchedulingMetricsScenario: func(c *testingclock.FakeClock, pod *v1.Pod) (*framework.QueuedPodInfo, error) {
+				// Create a queue with PreEnqueuePlugin
+				m := map[string][]framework.PreEnqueuePlugin{"": {&preEnqueuePlugin{allowlists: []string{"foo"}}}}
+				queue := NewTestQueue(ctx, newDefaultQueueSort(), WithClock(c), WithPreEnqueuePluginMap(m), WithPluginMetricsSamplePercent(0))
+
+				// Create a pod without PreEnqueuePlugin label.
+				gatedPod := st.MakePod().Name("gated-test-pod").Namespace("test-ns").UID("test-uid").Obj()
+				err := queue.Add(logger, gatedPod)
+				if err != nil {
+					t.Fatalf("Failed to add a pod %v", err)
+				}
+				// Check pod is added to the unschedulablePods queue.
+				if getUnschedulablePod(queue, gatedPod) != gatedPod {
+					t.Errorf("Pod %v was not found in the unschedulablePods.", gatedPod.Name)
+				}
+				// Override clock to get different InitialAttemptTimestamp
+				c.Step(1 * time.Minute)
 
-	// Case 4: A gated pod is created and scheduled after lifting gate. The queue operations are
-	//  Add gated pod -> check unschedulablePods -> lift gate & update pod -> Pop.
-	c = testingclock.NewFakeClock(timestamp)
-	// Create a queue with PreEnqueuePlugin
-	m := map[string][]framework.PreEnqueuePlugin{"": {&preEnqueuePlugin{allowlists: []string{"foo"}}}}
-	queue = NewTestQueue(ctx, newDefaultQueueSort(), WithClock(c), WithPreEnqueuePluginMap(m), WithPluginMetricsSamplePercent(0))
+				// Update pod with the required label to get it out of unschedulablePods queue.
+				updateGatedPod := gatedPod.DeepCopy()
+				updateGatedPod.Labels = map[string]string{"foo": ""}
+				queue.Update(logger, gatedPod, updateGatedPod)
 
-	// Create a pod without PreEnqueuePlugin label.
-	gatedPod := st.MakePod().Name("gated-test-pod").Namespace("test-ns").UID("test-uid").Obj()
-	err = queue.Add(logger, gatedPod)
-	if err != nil {
-		t.Fatalf("Failed to add a pod %v", err)
-	}
-	// Check pod is added to the unschedulablePods queue.
-	if getUnschedulablePod(queue, gatedPod) != gatedPod {
-		t.Errorf("Pod %v was not found in the unschedulablePods.", gatedPod.Name)
+				return queue.Pop()
+			},
+			wantAttempts:         1,
+			wantInitialAttemptTs: timestamp.Add(1 * time.Minute),
+		},
 	}
-	// Override clock to get different InitialAttemptTimestamp
-	c.Step(1 * time.Minute)
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
 
-	// Update pod with the required label to get it out of unschedulablePods queue.
-	updateGatedPod := gatedPod.DeepCopy()
-	updateGatedPod.Labels = map[string]string{"foo": ""}
-	queue.Update(logger, gatedPod, updateGatedPod)
+			c := testingclock.NewFakeClock(timestamp)
 
-	pInfo, err = queue.Pop()
-	if err != nil {
-		t.Fatalf("Failed to pop a pod %v", err)
+			podInfo, err := test.perPodSchedulingMetricsScenario(c, pod)
+			if err != nil {
```

Comment:
if we create the queue here, then we can also call queue.pop here since all test cases does it at the end, this helps simplify the test cases.

Reply:
Addressed.